### PR TITLE
Create test route task

### PR DESCRIPTION
**Pull request by Mohammed-Sbeata**

### CODE REVIEW

File: __tests__/task.test.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import supertest from 'supertest';
+import dotenv from 'dotenv';
+import {
+  afterAll, beforeEach, describe, test, expect,
+} from '@jest/globals';
```
Comment (nada-abuzaid) [inline __tests__/task.test.ts:5]: Jest puts each of these methods into the global environment so u don't have to import anything to use them.

File: __tests__/task.test.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      });
+  });
+
+  test('Should return 200 when add task', (done) => {
+    supertest(app)
+      .post('/project/10/task')
+      .set('cookie', `token=${process.env.TOKEN}`)
+      .send({
+        title: 'add query tasks',
+      })
```
Comment (nada-abuzaid) [inline __tests__/task.test.ts:59]: read the `taskSchema` again and update the request body here.

File: __tests__/task.test.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      .expect('Content-Type', 'application/json; charset=utf-8')
+      .end((err) => {
+        if (err) return done(err);
+        done();
+      });
+  });
+
+  test('Should return 200 when add task', (done) => {
+    supertest(app)
+      .post('/project/10/task')
```
Comment (nada-abuzaid) [inline __tests__/task.test.ts:55]: do u have project id 10 in ur database? if yes, do u have a manager role? please check to pass the test.

File: __tests__/task.test.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      .set('cookie', `token=${process.env.TOKEN}`)
+      .send({
+        title: 'add query tasks',
+      })
+      .expect(200)
```
Comment (nada-abuzaid) [inline __tests__/task.test.ts:60]: 200 or 201?

File: __tests__/task.test.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+      .end((err, res) => {
+        if (err) return done(err);
+        expect(res.body.message).toEqual('Task added successfully');
+        expect(res.body.data).toBe('array');
```
Comment (nada-abuzaid) [inline __tests__/task.test.ts:64]: `res.body.data` does this return a text  with value "array"?

File: src/database/config/buildDb.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+const buildDb = async () => {
+  const buildFile = readFileSync(join(__dirname, 'build.sql')).toString();
+  const fakeDataFile = readFileSync(join(__dirname, 'fakeData.sql')).toString();
+  try {
+    await connection.query(buildFile + fakeDataFile);
+  } catch (err) {
+    console.log('DB Error: ', err);
+  }
+};
```
Comment (nada-abuzaid) [inline src/database/config/buildDb.ts:13]: Why did you create this file? Khaled created it already, delete it please

**Review by nada-abuzaid — CHANGES_REQUESTED** [review]

File: __tests__/task.test.ts (hunk 3 above)
Comment (Mohammed-Sbeata) [reply __tests__/task.test.ts:55]: yes, I'm sure

File: src/database/config/buildDb.ts (hunk 6 above)
Comment (Mohammed-Sbeata) [reply src/database/config/buildDb.ts:13]: ok, deleted it.

File: __tests__/task.test.ts (hunk 5 above)
Comment (Mohammed-Sbeata) [reply __tests__/task.test.ts:64]: sorry, without single quote or deleted it

File: __tests__/task.test.ts (hunk 4 above)
Comment (Mohammed-Sbeata) [reply __tests__/task.test.ts:60]: 201 because it created or added task.

File: __tests__/task.test.ts (hunk 1 above)
Comment (Mohammed-Sbeata) [reply __tests__/task.test.ts:5]: ok, thanks.

File: __tests__/task.test.ts (hunk 2 above)
Comment (Mohammed-Sbeata) [reply __tests__/task.test.ts:59]: ok, updated it.

**Pull request closed without merging**